app/app.go

```diff
@@ -103,6 +103,16 @@ func (a *App) InitServer() {
 		if a.IsLeader() {
 			a.srv.dndTask = model.CreateRecurringTask("Unset DND Statuses", a.UpdateDNDStatusOfUsers, 300*time.Second)
 		}
+		a.srv.AddClusterLeaderChangedListener(func() {
+			mlog.Info("Cluster leader changed. Determining if unset dnd status task should be running", mlog.Bool("isLeader", a.IsLeader()))
+			if a.IsLeader() {
+				a.srv.dndTask = model.CreateRecurringTask("Unset DND Statuses", a.UpdateDNDStatusOfUsers, 300*time.Second)
+			} else {
+				if a.srv.dndTask != nil {
+					a.srv.dndTask.Cancel()
```

Comment:
This is a tricky one, but you are "Cancelling" the dndTask and not setting it to nil, so... for example if you have 3 nodes, and the leadership changes from the node 1 to the node 2, the node one cancel the task and the node 2 starts it, then, if the leadership changes from node 2 to node 3, the node one tries to cancel the task again (closing an already closed channel).

Reply:
i think setting the task to nil in else if condition should take care of this
```go
else {
    if a.srv.dndTask != nil {
        a.srv.dndTask.Cancel()
        a.srv.dndTask = nil
    }
}
```
 what do you think?

Reply:
@jespino see comments above ^